### PR TITLE
Use deckardcain library to identify source code in legacy interface

### PR DESCRIPTION
**Pull request by honzajavorek**
- Using [Deckard Cain](https://github.com/apiaryio/deckardcain) for identifying ambiguous source code.
- Removed `newVersionRegExp` from the interface (will need additional changes in Apiary codebase once we will upgrade the core application to Fury version containing changes in this PR).
- Removed some tests as they were testing the same stuff which is already tested in Deckard's tests.

### CODE REVIEW

File: src/legacy/blueprint-parser.coffee [hunk 1]
```diff
@@ -109,6 +109,12 @@ getLocalAst = ({code, blueprintId, sourcemap, emitter}, cb) ->
       return cb err
     cb null, apiaryBlueprintAdapter.transform apiaryAst
 
+  else
+    if mediaType
```
Comment (ttahmouch) [inline src/legacy/blueprint-parser.coffee:113]: @honzajavorek 
Is this state even possible?

File: src/legacy/blueprint-parser.coffee [hunk 2]
```diff
@@ -120,5 +126,4 @@ getAstSync = (code) ->
 module.exports = {
   parseApiaryBlueprintSync: getAstSync
   parse: getLocalAst
-  newVersionRegExp: NEW_VERSION_REGEXP # layout control control (editor)
```
Comment (ttahmouch) [inline src/legacy/blueprint-parser.coffee:123]: @honzajavorek 
Would there be any reason for us to want to leave the `RegExp` exported for `apiary`, or should we use `deckardcain` in `apiary`?

**Comment by ttahmouch** [comment]
@honzajavorek 
LGTM. I probably shouldn't be an `assignee` for `fury.js` though.

File: src/legacy/blueprint-parser.coffee (hunk 1 above)
Comment (honzajavorek) [inline src/legacy/blueprint-parser.coffee:113]: Not now, but in future, DC could return e.g. `application/swagger+json`

File: src/legacy/blueprint-parser.coffee (hunk 2 above)
Comment (honzajavorek) [inline src/legacy/blueprint-parser.coffee:123]: No reason. Should be encapsulated, IMHO. We should use DC in Apiary if we need this.

**Comment by zdne** [comment]
@honzajavorek Per our discussion, I am surprised this is in the Fury.js at all! I believed the decision was to remove the "legacy" interface from Fury. Is this no longer the case?

**Comment by zdne** [comment]
Hey @honzajavorek, I 've discussed this briefly with @Baggz and @freaz and we have agreed that it would be the best to do all the work that leads towards isolation of "legacy" interface on a separate branch. 

Once that integration branch is completely ready let's open a PR to master and release as needed.

Until then I suggest to reopen this PR on a shared/integration branch

**Pull request closed without merging**